google-cloud-spanner/src/test/java/com/google/cloud/spanner/connection/StatementParserTest.java

```diff
@@ -1266,6 +1266,156 @@ public void testPostgreSQLGetQueryParameters() {
+            .hasReturningClause());
+    assertTrue(
+        parser
+            .parse(Statement.of("insert into x (a,b) values (1,2)then/*comment*/return *"))
+            .hasReturningClause());
+    assertTrue(
+        parser
+            .parse(Statement.of("insert into x (a,b) values (1,2)then/*comment*/return(a)"))
+            .hasReturningClause());
+    assertTrue(
+        parser
+            .parse(Statement.of("insert into x (a,b) values (1,2)then /*comment*/return(a)"))
+            .hasReturningClause());
+    assertTrue(
+        parser
+            .parse(
+                Statement.of("insert into x (a,b) values (1,2)/*comment*/then/*comment*/return(a)"))
+            .hasReturningClause());
+    assertTrue(
+        parser
+            .parse(
+                Statement.of(
+                    "insert into x (a,b) values (1,2)/*comment*/then/*comment*/return/*comment*/(a)"))
+            .hasReturningClause());
+    assertTrue(
+        parser
+            .parse(
+                Statement.of(
+                    "insert into x (a,b) values (1,2)/*comment"
+                        + "*/then"
+                        + "/*comment"
+                        + "*/return/*"
+                        + "comment*/(a)"))
+            .hasReturningClause());
+    assertFalse(
+        parser
+            .parse(Statement.of("insert into x (a,b) values (1,2) returning (a)"))
+            .hasReturningClause());
+    assertFalse(
+        parser
+            .parse(Statement.of("insert into x (a,b) values (1,2) /*then return **/"))
+            .hasReturningClause());
+    assertFalse(
+        parser.parse(Statement.of("insert into x (a,b) values (1,2)")).hasReturningClause());
+    assertFalse(
+        parser
+            .parse(Statement.of("insert into x (a,b) values (1,2)thenreturn*"))
+            .hasReturningClause());
+  }
+
+  @Test
+  public void testPostgreSQLReturningClause() {
+    assumeTrue(dialect == Dialect.POSTGRESQL);
+
+    PostgreSQLStatementParser parser = (PostgreSQLStatementParser) this.parser;
+    assertTrue(
+        parser
+            .parse(Statement.of("insert into x (a,b) values (1,2) returning *"))
+            .hasReturningClause());
+    assertTrue(
+        parser
+            .parse(Statement.of("insert into x (a,b) values (1,2)returning *"))
+            .hasReturningClause());
+    assertTrue(
+        parser
+            .parse(Statement.of("insert into x (a,b) values (1,2) returning(a)"))
+            .hasReturningClause());
+    assertTrue(
+        parser
+            .parse(Statement.of("insert into x (a,b) values (1,2)returning(a)"))
+            .hasReturningClause());
+    assertTrue(
+        parser
+            .parse(Statement.of("insert into x (a,b) values (1,2)/*comment*/returning(a)"))
+            .hasReturningClause());
+    assertTrue(
+        parser
+            .parse(
+                Statement.of("insert into x (a,b) values (1,2)/*comment*/returning/*comment*/(a)"))
```

Comment:
We also need tests with nested comments (see also my comment on the parser itself)

Reply:
I've already added test with nested comments. Responded on the above comment.